implementation/src/rpc-proxy/verifiedProvider.ts

```diff
@@ -89,17 +160,30 @@ export class VerifiedProvider {
     const vm = await VM.create({ common: this.common });
 
     await vm.stateManager.checkpoint();
-    // TODO: use batching to speed this up
-    for (let access of accessList) {
-      const proof = await this.getProof(
-        access.address,
-        access.storageKeys,
-        blockNumber,
-      );
-
-      const { nonce, balance, codeHash, storageProof: storageAccesses } = proof;
-      const code = await this.getCode(access.address, blockNumber, codeHash);
-      const address = AddressEthereumJs.fromString(access.address);
+
+    const responses = chunk(
+      await this.fetchRequestsInBatches(
+        flatten(accessList.map((access) => {
+          return [
+            {
+              type: 'account',
+              blockNumber,
+              storageSlots: access.storageKeys,
+              address: access.address,
+            },
+            {
+              type: 'code',
+              blockNumber,
+              address: access.address,
+            }, 
+          ]
+        }
+      )), REQUEST_BATCH_SIZE
+    ), 2);
+
+    for (const [accountProof, code] of responses as [AccountResponse, CodeResponse][]) {
```

Comment:
Rename `responses` to `tupleResponses`?

```suggestion
    const tupleResponses = chunk(
      await this.fetchRequestsInBatches(
        flatten(accessList.map((access) => {
          return [
            {
              type: 'account',
              blockNumber,
              storageSlots: access.storageKeys,
              address: access.address,
            },
            {
              type: 'code',
              blockNumber,
              address: access.address,
            }, 
          ]
        }
      )), REQUEST_BATCH_SIZE
    ), 2);

    for (const [accountProof, code] of tupleResponses as [AccountResponse, CodeResponse][]) {
```